echopype/calibrate/cal_params.py

```diff
@@ -297,13 +297,13 @@ def get_vend_cal_params_power(beam: xr.Dataset, vend: xr.Dataset, param: str) ->
     idxmin = idxmin.where(~transmit_isnull, 0).astype(int)
 
     # Get param dataarray into correct shape
-    da_param = (
-        vend[param]
-        .expand_dims(dim={"ping_time": idxmin["ping_time"]})  # expand dims for direct indexing
-        .sortby(idxmin.channel)  # sortby in case channel sequence differs in vend and beam
-    )
+    da_param = vend[param].transpose("pulse_length_bin", "channel")
+
+    if not np.array_equal(da_param.channel.data, idxmin.channel.data):
+        da_param = da_param.sortby(
+            da_param.channel, ascending=False
+        )  # sortby because channel sequence differs in vend and beam
```

Comment:
@lsetiawan : from our discussion last Friday: 
The `ping_time` dimension gets removed here, and the tests probably pass because that dimension =1 in those tests. Could you look into this? I would suggest making the `ping_time` much much much longer, as the performance impact would likely show up there, and as you know a long `ping_time` dimension is the most common scenario with the data.